# Importing the Inventory

In [1]:
import json

In [2]:
fd= open("record.json",'r')
r=fd.read()
fd.close()

record= json.loads(r)

In [3]:
record

{'1000': {'Name': 'Chococake',
  'price': 20,
  'Quantity': 40,
  'Company': 'zzz',
  'Best_before': '3 months',
  'Category': 'pastry'},
 '1001': {'Name': 'Vanillacake',
  'price': 20,
  'Quantity': 40,
  'Company': 'zzz',
  'Best_before': '3 months',
  'Category': 'pastry'},
 '1002': {'Name': 'Pistacake',
  'price': 20,
  'Quantity': 40,
  'Company': 'zzz',
  'Best_before': '3 months',
  'Category': 'pastry'},
 '1003': {'Name': 'plumcake',
  'price': 20,
  'Quantity': 40,
  'Company': 'zzz',
  'Best_before': '3 months',
  'Category': 'pastry'},
 '1004': {'Name': 'Bananacake',
  'price': 20,
  'Quantity': 40,
  'Company': 'zzz',
  'Best_before': '3 months',
  'Category': 'pastry'},
 '1005': {'Name': 'Coke',
  'price': 30,
  'Quantity': 50,
  'Company': 'zzz',
  'Best_before': '5 months',
  'Category': 'drinks'},
 '1006': {'Name': 'Fanta',
  'price': 30,
  'Quantity': 50,
  'Company': 'zzz',
  'Best_before': '5 months',
  'Category': 'drinks'},
 '1007': {'Name': 'Limca',
  'price': 30,

# Purchase of products

In [4]:
prod_ui=str(input('enter the product Id: '))
prod_quan=int(input('enter the quantity of the product: '))

print('Product name:',record[prod_ui]['Name'])
print('Product price:',record[prod_ui]['price'])
print('Amount to be paid:',record[prod_ui]['price']*prod_quan)
print('quantity before purchase:',record[prod_ui]['Quantity'])
print('quantity left:',record[prod_ui]['Quantity']-prod_quan)

record[prod_ui]['Quantity']=record[prod_ui]['Quantity']-prod_quan

enter the product Id: 1031
enter the quantity of the product: 20
Product name: thumbsup
Product price: 40
Amount to be paid: 800
quantity before purchase: 100
quantity left: 80


### we can see the quantity before purchase and after purchase

In [5]:
record

{'1000': {'Name': 'Chococake',
  'price': 20,
  'Quantity': 40,
  'Company': 'zzz',
  'Best_before': '3 months',
  'Category': 'pastry'},
 '1001': {'Name': 'Vanillacake',
  'price': 20,
  'Quantity': 40,
  'Company': 'zzz',
  'Best_before': '3 months',
  'Category': 'pastry'},
 '1002': {'Name': 'Pistacake',
  'price': 20,
  'Quantity': 40,
  'Company': 'zzz',
  'Best_before': '3 months',
  'Category': 'pastry'},
 '1003': {'Name': 'plumcake',
  'price': 20,
  'Quantity': 40,
  'Company': 'zzz',
  'Best_before': '3 months',
  'Category': 'pastry'},
 '1004': {'Name': 'Bananacake',
  'price': 20,
  'Quantity': 40,
  'Company': 'zzz',
  'Best_before': '3 months',
  'Category': 'pastry'},
 '1005': {'Name': 'Coke',
  'price': 30,
  'Quantity': 50,
  'Company': 'zzz',
  'Best_before': '5 months',
  'Category': 'drinks'},
 '1006': {'Name': 'Fanta',
  'price': 30,
  'Quantity': 50,
  'Company': 'zzz',
  'Best_before': '5 months',
  'Category': 'drinks'},
 '1007': {'Name': 'Limca',
  'price': 30,

### Here in updated record we can see 1031 id product's quantity has decreased from 100 to 80

In [6]:
jss= json.dumps(record)

# Now lets us update the Inventory  

In [7]:
fd= open("record.json",'w')
w=fd.write(jss)
fd.close()

In [8]:
fd= open("record.json",'r')
r=fd.read()
fd.close()

updated_record= json.loads(r)

# The updated Inventory 

In [9]:
updated_record

{'1000': {'Name': 'Chococake',
  'price': 20,
  'Quantity': 40,
  'Company': 'zzz',
  'Best_before': '3 months',
  'Category': 'pastry'},
 '1001': {'Name': 'Vanillacake',
  'price': 20,
  'Quantity': 40,
  'Company': 'zzz',
  'Best_before': '3 months',
  'Category': 'pastry'},
 '1002': {'Name': 'Pistacake',
  'price': 20,
  'Quantity': 40,
  'Company': 'zzz',
  'Best_before': '3 months',
  'Category': 'pastry'},
 '1003': {'Name': 'plumcake',
  'price': 20,
  'Quantity': 40,
  'Company': 'zzz',
  'Best_before': '3 months',
  'Category': 'pastry'},
 '1004': {'Name': 'Bananacake',
  'price': 20,
  'Quantity': 40,
  'Company': 'zzz',
  'Best_before': '3 months',
  'Category': 'pastry'},
 '1005': {'Name': 'Coke',
  'price': 30,
  'Quantity': 50,
  'Company': 'zzz',
  'Best_before': '5 months',
  'Category': 'drinks'},
 '1006': {'Name': 'Fanta',
  'price': 30,
  'Quantity': 50,
  'Company': 'zzz',
  'Best_before': '5 months',
  'Category': 'drinks'},
 '1007': {'Name': 'Limca',
  'price': 30,

# Billing The purchase of products

In [10]:
sales={1:{'pdt':prod_ui,'quantity':prod_quan,'Amount':record[prod_ui]['price']*prod_quan},
      2:{'pdt':prod_ui,'quantity':prod_quan,'Amount':record[prod_ui]['price']*prod_quan}
     }

In [11]:
sales

{1: {'pdt': '1031', 'quantity': 20, 'Amount': 800},
 2: {'pdt': '1031', 'quantity': 20, 'Amount': 800}}

In [12]:
jsa= json.dumps(sales)

In [13]:
jsa

'{"1": {"pdt": "1031", "quantity": 20, "Amount": 800}, "2": {"pdt": "1031", "quantity": 20, "Amount": 800}}'

### Now lets load this data into another file named sales.json 

In [15]:
fd= open("sales.json",'w')
w=fd.write(jsa)
fd.close

sale_record= json.dumps(w)